pkg/accounts/adaptor/repository/prisma.ts

```diff
@@ -0,0 +1,251 @@
+import { Option, Result } from '@mikuroxina/mini-fn';
+import { type PrismaClient } from '@prisma/client';
+
+import type { ID } from '../../../id/type.js';
+import type {
+  AccountFrozen,
+  AccountID,
+  AccountName,
+  AccountRole,
+  AccountSilenced,
+  AccountStatus,
+} from '../../model/account.js';
+import { Account } from '../../model/account.js';
+import type {
+  AccountRepository,
+  AccountVerifyTokenRepository,
+} from '../../model/repository.js';
+
+interface AccountPrismaArgs {
+  id: string;
+  name: string;
+  nickname: string;
+  mail: string;
+  passphrase_hash: string | null;
+  bio: string;
+  role: number;
+  frozen: number;
+  silenced: number;
+  status: number;
+  created_at: Date;
+  updated_at: Date | null;
+  deleted_at: Date | null;
+}
+
+export class PrismaAccountRepository implements AccountRepository {
+  readonly prisma: PrismaClient;
+
+  constructor(prisma: PrismaClient) {
+    this.prisma = prisma;
+  }
+
+  async create(account: Account): Promise<Result.Result<Error, void>> {
+    try {
+      await this.prisma.account.create({
+        data: {
+          ...this.toPrismaArgs(account),
+        },
+      });
+    } catch (e) {
+      return Result.err(e as Error);
+    }
+
+    return Result.ok(undefined);
+  }
+
+  async findByMail(mail: string): Promise<Option.Option<Account>> {
+    try {
+      const res = await this.prisma.account.findUnique({
+        where: {
+          mail: mail,
+        },
+      });
+      if (!res) {
+        return Option.none();
+      }
+      return Option.some(this.fromPrismaArgs(res));
+    } catch (e) {
+      return Option.none();
+    }
```

Comment:
```suggestion
    const res = await this.prisma.account.findUnique({
      where: {
        mail: mail,
      },
    });
    if (!res) {
      return Option.none();
    }
    return Option.some(this.fromPrismaArgs(res));
```
`findUnique` からの例外を `none` として扱って握りつぶすのは, プログラミングミスなどを起因とするエラーが見えなくなるのでよくないと思います. 他の `find` 系メソッドの実装も同様です.